src/main/java/core/basesyntax/strategy/impl/SupplyOperation.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.HashMap;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void handle(HashMap<String, Integer> storage, FruitTransaction transaction) {
+        String fruit = transaction.getFruit();
+        int quantity = transaction.getQuantity();
+        if (quantity < 0) {
+            throw new RuntimeException(String.format("Error supplying "
+                    + "fruit %s: quantity %d of supply is less than zero", fruit, quantity));
+        }
+        storage.put(fruit, (storage.get(fruit) + quantity));
```

Comment:
There's a potential `NullPointerException` here if the `storage.get(fruit)` returns `null`. You should check if the fruit already exists in the storage and handle the case where it does not. Consider using `storage.getOrDefault(fruit, 0)` to avoid this issue.